src/librustc/ty/maps/config.rs

```diff
@@ -73,6 +73,12 @@ impl<'tcx> QueryDescription<'tcx> for queries::normalize_ty_after_erasing_region
     }
 }
 
+impl<'tcx> QueryDescription<'tcx> for queries::evaluate_obligation<'tcx> {
+    fn describe(_tcx: TyCtxt, goal: CanonicalPredicateGoal<'tcx>) -> String {
+        format!("evaluating trait selection obligation `{:?}`", goal)
```

Comment:
So the failing test `impl-trait/auto-trait-leak.rs` is now including this message in its output. In general, `{:?}` should be avoided here, as this a message that may be shown to the user. That said, nicely formatting this message could be kind of annoying. I guess something like this might be good enough for now:

```
format!("evaluating trait selection obligation `{}`", goal.value.value)
```

I'm trying this locally. This will just skip over the canonical binders and the param-env.